### PR TITLE
[RFR] Meteor Webpack React Redux

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+Meteor with Webpack, React and Redux
+===================
+
+# Introduction
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: Use h2 (`##`) instead

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+- Uses its own building tool which is slower than webpack and is not as extensible
+- Forces you to use global variables and old school namespaces
+- **es6** recent integration does not include `import` and `export`
+
+*MDG* (the company behind Meteor) is aware of those problems and is working on it as seen on their forum [here](https://forums.meteor.com/t/next-steps-on-blaze-and-the-view-layer/13561). Fortunately, some Meteor developers have worked on ways to improve this.
+
+# How can we use those technologies together
+I'm aware of two projects trying to address the Meteor/Webpack problem:
+
+1. https://github.com/thereactivestack/kickstart by **eXon**
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: Add hyperlink on the lib author name to twitter account?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+### Redux actions
+We write the authentication related actions in `./app/client/actions/auth.js`.
+
+#### redux action for google signin
+Meteor expose the `loginWithGoogle` method which will take care of the oauth process with a popup.
+
+``` js
+export function loginWithGoogle() {
+	return () => {
+		Meteor.loginWithGoogle(err => {
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: is `Meteor` a global here, or does the use have to include a dependency?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+# Redux
+Making **Meteor** work with **Redux** is pretty easy. Let's begin with a simple example: authentication with google.
+
+### Configure Meteor authentication
+Proceed as usual:
+
+  - add the following Meteor packages: `accounts-google`, `service-configuration`
+  - get the api key and secret from your google console and put them in your Meteor settings file (`settings.json`)
+    ``` json
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: I don't think this is valid markdown for an indented codeblock. Make sure it translates to HTML correctly.

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+			if (err) {
+				alert('Error while login with google');
+			}
+		});
+	};
+}
+```  
+We don't handle the successful result of the action here as the current user will be automatically available with `Meteor.user()`  which is a reactive datasource.
+
+#### redux action to load the user
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: capital R

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+Make sure you installed the redux [thunk-middleware](https://github.com/gaearon/redux-thunk) for this.
+
+``` js
+export const USER_LOGGING_IN = 'USER_LOGGING_IN';
+export const USER_DATA = 'USER_DATA';
+
+export function loadUser() {
+  return dispatch => {
+    Tracker.autorun(() => {
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: What's `Tracker`?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+        type: 'USER_DATA',
+        data: Meteor.user(),
+      });
+    });
+  };
+}
+```
+As we are running the dispatch inside a `Tracker.autorun`, the `USER_LOGGING_IN` action will be dispatched everytime the result of the `Meteor.loggingIn()` computation change. The same will happen for `Meteor.user()`.
+
+#### redux action to logout
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: s/redux/Redux

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+export function logout() {
+    return () => {
+        Meteor.logout();
+    };
+}
+```
+
+### Redux reducers
+
+Let's make the data available in the redux state with reducers:
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: s/in the/to the

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+
+  default:
+    return state;
+  }
+}
+```
+
+### React
+
+You probably want the user to be available everywhere so we need to ensure `Meteor.user()` is bound to our store right from the beginning. If you used **jedwards** project, this is what the `main_client.js` look like:
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: s/to our store/to the store

Not a big fan of the "we" in blog posts

**Comment by fzaninotto** [comment]
Will you be providing a link to a repo with the complete source?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md (hunk 4 above)
Comment (djhi) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: It does render correctly on github at least

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md (hunk 3 above)
Comment (djhi) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: I added an explanation about Meteor global in previous section

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md (hunk 2 above)
Comment (djhi) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: I can't find any account for jedwards

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md (hunk 4 above)
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: You have to check the Jekyll rendered version

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md (hunk 6 above)
Comment (djhi) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: I added an explanation about Tracker below this code section

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 10]
```diff
@@ -0,0 +1,397 @@
+Meteor with Webpack, React and Redux
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: You should add something less generic, like "... in practice", or "...It works, here is the proof"

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 11]
```diff
@@ -0,0 +1,453 @@
+---
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: Could you rename the article to have the date changed to today?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 12]
```diff
@@ -0,0 +1,453 @@
...
+title: "Meteor with Webpack, React and Redux"
+excerpt: "Making Meteor works with Webpack, React and Redux."
+image: /images/blog/leankanban.jpg
+author: <a href="https://twitter.com/gildaspk">Gildas</a>
+tags:
+- Meteor
+- React
+- Redux
+- Webpack
+---
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: JavaScript ?

File: blog/_posts/2015-11-24-meteor-webpack-react-redux.md [hunk 13]
```diff
@@ -0,0 +1,453 @@
...
+    loginWithGoogle: PropTypes.func.isRequired,
+};
+
+export default Home;
+```
+
+## Going further
+
+We only scratched the surface about how to make Meteor and Redux work together here. I've been working on a real application using this method. This is a Trello like application customized to handle nutritional plannings and to provide nutritionist coaches a tool to coach their clients.
+
```
Comment (fzaninotto) [inline blog/_posts/2015-11-24-meteor-webpack-react-redux.md]: Could you add a screenshot of the app here ?

**Pull request merged**